File: src/main/java/com/CantvasAngular/CantvasAngular/Models/Assignment.java
package com.CantvasAngular.CantvasAngular.Models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinTable;
import jakarta.persistence.*;

@Entity
public class Assignment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String name;
    String document;
    Integer maxValue;

    @ManyToOne
    Student student;

    public Assignment() {
    }

    public Assignment(String name, String document) {
        this.name = name;
        this.document = document;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDocument() {
        return document;
    }

    public void setDocument(String document) {
        this.document = document;
    }
}

File: angularclient/src/app/model/assignment.ts
export class Assignment {
  id: number;
  name: string;
  document: string;
  maxValue: number;

  constructor(id: number, name: string, document: string, maxValue: number) {
    this.id = id;
    this.name = name;
    this.document = document;
    this.maxValue = maxValue;
  }
}
